src/services/ui/src/pages/actions/WithdrawRai.tsx

```diff
@@ -1,44 +1,183 @@
+        <ActionFormIntro title={"Withdraw Formal RAI Response Details"}>
+          <RequiredIndicator /> Indicates a required field
+          <p className="font-light mb-6 max-w-4xl">
+            Complete this form to withdraw the Formal RAI response. Once
+            complete, you and CMS will receive an email confirmation.
+          </p>
+        </ActionFormIntro>
+        {/* Package ID and type info */}
+        <PackageInfo item={item} />
+        {/* Attachments */}
+        <h3 className="font-bold text-2xl font-sans">Attachments</h3>
+        <AttachmentsSizeTypesDesc
+          faqLink={"/faq/#medicaid-spa-rai-attachments"}
+        />
+        {attachments.map(({ name, label, required }) => (
+          <FormField
+            key={String(name)}
+            control={form.control}
+            name={`attachments.${String(name)}` as Path<typeof schema>}
+            render={SlotAttachments({
+              label: (
+                <>
+                  {label}
+                  {required ? <RequiredIndicator /> : ""}
+                </>
+              ),
+              message: <FormMessage />,
+              className: "my-4",
+            })}
+          />
+        ))}
+        {/* Additional Info */}
+        <FormField
+          control={form.control}
+          name={"additionalInformation" as Path<typeof schema>}
+          render={SlotAdditionalInfo({
+            label: <p>Explain your need for withdrawal.</p>,
+            description: "4,000 characters allowed",
+            className: "pt-6",
+            required: true,
+          })}
+        />
+        {/* Error banner */}
+        {Object.keys(form.formState.errors).length !== 0 && (
+          <Alert className="my-6" variant="destructive">
+            Input validation error(s)
+            <ul className="list-disc">
+              {Object.values(form.formState.errors).map(
+                (err, idx) =>
+                  err?.message && (
+                    <li className="ml-8 my-2" key={idx}>
+                      {err.message as string}
+                    </li>
+                  )
+              )}
+            </ul>
+          </Alert>
+        )}
+        {/* Pre-submit message banner */}
+        <Alert variant={"infoBlock"} className="my-2 w-full flex-row text-sm">
+          <Info />
+          <p className="ml-2">
+            Once complete, you and CMS will receive an email confirmation.
+          </p>
+        </Alert>
+        {/* Buttons */}
+        <div className="flex gap-2 my-8">
+          <Button
+            type={"button"}
+            onClick={() => {
+              setModalContent({
+                header: "Withdraw RAI response?",
+                body: `The RAI response for ${item._source.id} will be withdrawn, and CMS will be notified.`,
+                acceptButtonText: "Yes, withdraw response",
+                cancelButtonText: "Cancel",
+              });
+              setModalOnAccept(() => confirmOnAccept);
```

Comment:
I haven't tested yet, but this looks like it needs to be updated to:

```js
setModalOnAccept(confirmOnAccept);
```

Reply:
This works as-intended. To properly test/confirm, change the `() => confirmOnAccept` to `confirmOnAccept` and try the Cancel button on this, or the other, forms.